keps/sig-autoscaling/20190307-configurable-scale-velocity-for-hpa.md

```diff
@@ -382,57 +424,57 @@ It may be defined by command line option `--horizontal-pod-autoscaler-downscale-
 
 The following API changes are needed:
 
-We should add `Scale{Up,Down}{Percent,Pods}` fields into the HPA spec
+- `HorizontalPodAutoscalerBehavior` should be added as a field to the HPA spec. It will contain fields
+  which describe the scaling behavior for both scale up and scale down. In the future other aspects of
+  the scaling behavior could be customized by adding fields here.
+- `HPAScaleConstraint` specifies the maximum change in the number of pods while scaling up or down
+  during autoscaling.
 
-The resulting solution will look like this:
+The resulting data structures will look like this:
 
 ```golang
-type HPAScaleConstraintValue struct {
-    Rate         *HPAScaleConstraintRateValue
-    DelaySeconds *int32
-
-type HPAScaleConstraintRateValue struct {
-    Pods          *int32
-    Percent       *int32
-    PeriodSeconds *int32
+type HPAScalingPolicyType string
+
+const (
+  PercentPolicy HPAScalingPolicyType = "percent"
+  PodsPolicy    HPAScalingPolicyType = "pods"
+)
+
+type HPAScalingPolicy struct {
+  Type          HPAScalingPolicyType
+  Value         int32
+  PeriodSeconds int32
+}
+
+type HPAScalingRules struct {
+  StabilizationWindowSeconds *int32
```

Comment:
I think we were going to add `Tolerance`, even if it's just in the KEP and not implemented yet.

Reply:
In the `non-goals` I mention tolerance as one of the non-goals.